commonjs plugin needs to come after babel plugin

dagda1 opened the pull request:
While looking at #680 I discovered that the `commonjs` needs to follow the babel plugin.

This PR moves it down:

```ts
      babel({
        babelHelpers: 'runtime',
        extensions: ['.js', '.ts'],
        presets: ['@babel/preset-env', '@babel/preset-typescript'],
        plugins: ['@babel/plugin-transform-runtime']
      }),
      commonjs(),
```

Commit:
commonjs plugin needs to come after babel plugin

Commit:
ad changeset

cowboyd commented:
@dagda1 at least it seems that the ubuntu and windows CLI tests are failing with a consistent error message 🤷🏻‍♂️

Commit:
fix extensions for node resolve and babel rollup plugins

dagda1 commented:
@cowboyd I fired this and a couple of other PRs through on Friday without due diligence.  Apologies for that. I'm going to pick them off one by one and take a bit more care.  Not going to rush

I do think this is the correct thing here though in this PR and I've done some more things in the last commit so please have a good look.

- Removed `@babel/preset-typescript`, we should have removed this when we added `rollup-plugin-typescript2`.
- Added a fuller extension list for babel and the `@rollup/plugin-node-resolve` plugins.

I'm going to knock off the PRs I created with a bit more care and one at a time.  I think they all have a good reason but I just fired them off and I should know better at my age :)

Commit:
change error assertion to check Error instanceof

Commit:
set sourceType for babel

Commit:
remove console.log

taras reviewed (approved):
Great catch

cowboyd reviewed (approved):
@dagda1 were you able to test this on the runner in stripes-components?

dagda1 commented:
@cowboyd I've had a quick hack by adding the preview package and I had to add a resolution to the main package.json in `stripes-components` 

`"@bigtest/bundler": "0.11.0-74609361",`

I also added this `babel.conifg.js` to stripes also

```json
module.exports = {
  presets: [
    [
      "@babel/preset-react"
    ]
  ]
}
```

The rollup babel plugin merges any babel file it finds with our config in the bundler

The jsx error certainly goes away once I do the above but we would need a proper `Mount` function to test it fully but I am fairly sure it will work now.  

I'm almost tempted to add `"@babel/preset-react"` to our babel-configuration.

Give me a shout tomorrow or anytime if you want to go through it.

We don't need `'@babel/preset-typescript'` anymore now we have the rollup typescript plugin so I'm glad we can remove this.

I have some thoughts about our build configuration in the bundler.
